### ISSUE TITLE
Difficulty in avoiding the cmake Python package

### ISSUE THREAD

**Issue by wshanks** [scikit-build/scikit-build#829]
I am working on packaging [qiskit-aer](https://github.com/Qiskit/qiskit-aer) for [conda-forge](https://github.com/conda-forge/staged-recipes/pull/21404). One issue that I have run into is that qiskit-aer specifies `cmake` in `setup_requires` and scikit-build tries to install it with pip. There is not a cmake Python package in conda-forge (I guess because it just wraps the main cmake code), so I can't pre-install the conda-forge equivalent to prevent scikit-build from installing the cmake Python package like I can for other Python packages. Do you have any suggestions for how to deal with this?

For now, I am just patching the setup.py in qiskit-aer to remove the setup_requires since all the dependencies are specified in the recipe any way. Could there be a flag to prevent scikit-build from installing cmake? I see that setup_requires was just deprecated by scikit-build. I guess the replacement is `build-system.requires` in pyproject.toml and the same issue would be present in that case?

**Comment by henryiii** [comment]
This is solved in scikit-build-core; it dynamically declares a dependency based on if there's a sufficient cmake executable already present. Once we get a few more features in (probably early next year), we'll look into who can directly upgrade, and that will solve this problem.

Quick recommendation, though - `setup_requires` is deprecated by setuptools too (for several years now) and shouldn't be used. I'd strongly recommend removing it in setup.py - it's much better to put cmake in pyproject.toml's build requirements. Then you could work around it by not using an isolated environment (which conda already does for you).

I don't think we should touch setup_requires (I'm aware that we do a little bit now, but I'm against that too), and ask that projects avoid using it. It's really setuptools handling it and causing the problem here.

TL;DR: I'd recommend asking the package to remove `setup_requires`, and patch it out for now in the recipe until they do.

**Comment by wshanks** [comment]
Thanks, @henryiii, for the explanation!
